core/drivers/crypto/caam/acipher/caam_ecc.c

```diff
@@ -322,6 +322,70 @@ static TEE_Result do_gen_keypair(struct ecc_keypair *key, size_t key_size)
 	return ret;
 }
 
+/*
+ * Check if MES_REP to be sent in descriptor is 00.
+ * Only required in case message length and key size is more than 40bytes
+ * because of limitation of Class 2 context register size on i.MX8M series.
+ * Example case below:
+ * When we try to do signature with P384-SHA384, in this case the key size
+ * will be 48bytes and Message size will also be 48bytes.
+ * This will work only when we set MES_REP = 0 in descriptor,
+ * but in this case, we don't need the padding to be done on the message.
+ *
+ * @msg_length: Message Length in bytes
+ * @key_size: Key size in bytes
+ */
+static bool msg_mes_rep(size_t msg_length, size_t key_size)
+{
+	if (IS_ENABLED(CFG_NXP_CAAM_C2_CTX_REG_WA) && msg_length > 40 &&
+	    key_size > 40)
+		return true;
+	else
+		return false;
+}
+
+/*
+ * Check if padding is required on message to make it of same length
+ * as that of key size.
+ * Only required in case message length and key size is more than 40bytes
+ * because of limitation of Class 2 context register size on i.MX8M series.
+ * So this will be applicable on P384 and P521 ECC curves because these
+ * curves have key_size more than 40bytes.
+ *
+ * @msg_length: Message Length in bytes
+ * @key_size: Key size in bytes
+ */
+static bool padding_required(size_t msg_length, size_t key_size)
+{
+	if (IS_ENABLED(CFG_NXP_CAAM_C2_CTX_REG_WA) && msg_length > 40 &&
```

Comment:
How about:
``` C
return msg_mes_rep(msg_length, key_size) && msg_length < key_size;
```

It's shorter and also highlights the connection to `msg_mes_rep()`.

By the way, what if msg_length is larger than key_size, is that an error?

Reply:
Thanks